docs/design.md

```diff
@@ -0,0 +1,78 @@
+# Design
+
+## Architecture
+
+![](architecture.png)
+
+### Control plane - Kubernetes
+
+#### Operator
+
+The k4e operator is a Kubernetes operator that consists of two components:
+ - controller responsible for reconciling `EdgeDevice` and `EdgeDeployment` CRs;
+ - HTTP API that is used by the k4e agent to get expected configuration and to post heartbeat messages. See [HTTP API schema](http-api.md) for more details.
+
+#### Object Storage
+
+Object Storage is used to store files created by workloads on devices and uploaded using k4e built-in mechanism.
+
+Object Bucket Claims provided by OCS and Noobaa and accessible with like S3 buckets. 
+
+See [Data Upload](data-upload.md) for more details.
+
+#### CRDs
+
+Operator manages two kinds of CRs: 
```

Comment:
What do you think about linking to https://github.com/jakub-dzon/k4e-operator/tree/main/config/crd/bases both CRDs and https://github.com/jakub-dzon/k4e-operator/tree/main/config/samples?

Reply:
Done